### PR TITLE
add claim when person exists in store

### PR DESCRIPTION
**Pull request by DownGoat**

### CODE REVIEW

File: src/Equinor.ProCoSys.Auth/Authorization/ClaimsTransformation.cs [hunk 1]
```diff
@@ -55,41 +56,43 @@ public class ClaimsTransformation : IClaimsTransformation
...
             }
 
             if (!await _permissionCache.HasUserAccessToPlantAsync(plantId, userOid.Value))
             {
-                _logger.LogInformation($"----- {GetType().Name} early exit, not a valid plant for user");
+                _logger.LogInformation("----- {Name} early exit, not a valid plant for user", GetType().Name);
                 return principal;
             }
+            
+            AddPersonExistsClaim(claimsIdentity, proCoSysPerson.AzureOid);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Auth/Authorization/ClaimsTransformation.cs:95]: move to line 76

File: src/Equinor.ProCoSys.Auth/Authorization/ClaimsTransformation.cs [hunk 2]
```diff
@@ -101,10 +104,15 @@ public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
                 await AddUserDataClaimForAllRestrictionRolesToIdentityAsync(claimsIdentity, plantId, userOid.Value);
             }
 
-            _logger.LogInformation($"----- {GetType().Name} completed");
+            _logger.LogInformation("----- {Name} completed", GetType().Name);
             return principal;
         }
 
+        private static void AddPersonExistsClaim(ClaimsIdentity claimsIdentity, string azureOid)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Auth/Authorization/ClaimsTransformation.cs:111]: Extend ClaimsTransformationTests with 2 tests: assert that claim is added when user exists, and assert that claim not added when user don't exists

**Comment by janid1967** [comment]
@DownGoat is the plan to check the claim in VerifyOidInDbMiddleware and not create/handle command when exists?

**Comment by janid1967** [comment]
@DownGoat U must manually raise VersionPrefix in Equinor.ProCoSys.Auth.csproj (we will get conflict with my work)

**Comment by DownGoat** [comment]
> @DownGoat is the plan to check the claim in VerifyOidInDbMiddleware and not create/handle CreatePersonCommand when user exists?

Yes, we are checking multiple times against the db if the person exists. Want to introduce this claim to avoid checking db

File: src/tests/Equinor.ProCoSys.Auth.Tests/Authorization/ClaimsTransformationTests.cs [hunk 3]
```diff
@@ -323,6 +324,27 @@ public async Task TransformAsync_OnSecondPlant_ShouldClearAllClaimsForFirstPlant
...
+        {
+            var result = await _dut.TransformAsync(_principalWithOid);
+
+            Assert.IsTrue(result.Claims.IsPersonCreated(Oid.ToString()));
+        }
+        
+        [TestMethod]
+        public async Task TransformAsync_ShouldNotAddExistsClaimsFoPerson_WhenPersonNotExists()
+        {
+            _principalWithOid = new ClaimsPrincipal();
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.Auth.Tests/Authorization/ClaimsTransformationTests.cs:338]: The Arrange-part should be
``` 
           _localPersonRepositoryMock.GetAsync(Oid).Returns((ProCoSysPerson)null);
           _personCacheMock.GetAsync(Oid).Returns((ProCoSysPerson)null);
```

**Review by janid1967 — APPROVED** [review]
Approved with minor approvement in a test

File: src/Equinor.ProCoSys.Auth/Authorization/ClaimsExtensions.cs [hunk 4]
```diff
@@ -102,5 +102,11 @@ public static string TryGetEmail(this IEnumerable<Claim> claims)
             }
             return null;
         }
+        
+        public static bool IsPersonCreated(this IEnumerable<Claim> claims, string userOid)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Auth/Authorization/ClaimsExtensions.cs:106]: Rename to PersonExistsLocally?

**Pull request merged**